include/boost/json/detail/impl/number.ipp

```diff
@@ -299,6 +310,9 @@ loop:
                 }
                 if(*p == 'e' || *p == 'E')
                 {
+                    // if decimal point has not yet been seen, assume it's here
+                    // since 100e0 is equivalent to 100.0e0
+                    if (pos_ < 0) pos_ = dig_;
```

Comment:
The statement has to be on its own line
```
if (pos_ < 0)
    pos_ = dig_;
```
for two reasons: to be able to set a breakpoint, and to distinguish the lines in coverage reports.